### ISSUE TITLE
"Update" breaks on staging when content is sourced from prod.

### ISSUE THREAD

**Issue by Niicck** [cityofaustin/techstack#2311]
From Desi:
I made a change in Joplin, clicked "update," and got this error.
![image](https://user-images.githubusercontent.com/5839508/59222783-e9c38580-8b8f-11e9-99e5-ebb1f9cabe08.png)

To reproduce: 
1. go to https://joplin-staging.herokuapp.com/admin/pages/9/edit/
2. make a change
3. click update
What I expect: Same behavior as "publish" on other pages—exit out to listing view 
What happened: that error above :(

Nick's notes:
- Sourcing data from prod with natural primary/foreign keys does not fix this issue.
- This only affects pages that were originally created in prod and then loaded into staging. Articles that were created on staging will not have this problem.

**Comment by Niicck** [comment]
Note: this bug only affects content that was in "Draft" state in prod when it was transferred over.

**Comment by easherma** [comment]
I strongly suspect this has something to do with the pagerevisions table for a given page. 
Might be related: 
https://github.com/wagtail/wagtail/issues/703

**Comment by easherma** [comment]
Purging revisions seems risky/also raises similar problems:
https://stackoverflow.com/questions/53268081/how-does-one-purge-a-database-of-old-revisions

**Comment by easherma** [comment]
`Environment:


Request Method: POST
Request URL: http://localhost:8000/admin/pages/9/edit/

Django Version: 2.2.2
Python Version: 3.6.5
Installed Applications:
['base.apps.BaseConfig',
 'users',
 'api.apps.APIConfig',
 'wagtail.api.v2',
 'wagtail.embeds',
 'wagtail.sites',
 'wagtail.users',
 'wagtail.snippets',
 'wagtail.documents',
 'wagtail.images',
 'wagtail.search',
 'wagtail.admin',
 'wagtail.core',
 'wagtail.contrib.forms',
 'wagtail.contrib.redirects',
 'wagtail.contrib.styleguide',
 'wagtail_modeltranslation',
 'wagtail_modeltranslation.makemigrations',
 'wagtail_modeltranslation.migrate',
 'modelcluster',
 'taggit',
 'rest_framework',
 'corsheaders',
 'modeltranslation',
 'graphene_django',
 'django_extensions',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'wagtail.contrib.modeladmin',
 'webpack_loader',
 'dbbackup',
 'smuggler']
Installed Middleware:
['django.contrib.sessions.middleware.SessionMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'wagtail.core.middleware.SiteMiddleware',
 'wagtail.contrib.redirects.middleware.RedirectMiddleware']



Traceback:

File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/local/lib/python3.6/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  44.         response = view_func(request, *args, **kwargs)

File "/usr/local/lib/python3.6/site-packages/wagtail/admin/urls/__init__.py" in wrapper
  102.             return view_func(request, *args, **kwargs)

File "/usr/local/lib/python3.6/site-packages/wagtail/admin/decorators.py" in decorated_view
  34.             return view_func(request, *args, **kwargs)

File "/usr/local/lib/python3.6/site-packages/wagtail/admin/views/pages.py" in edit
  368.                 revision.publish()

File "/usr/local/lib/python3.6/site-packages/wagtail/core/models.py" in publish
  1504.         page = self.as_page_object()

File "/usr/local/lib/python3.6/site-packages/wagtail/core/models.py" in as_page_object
  1459.         obj = self.page.specific_class.from_json(self.content_json)

Exception Type: AttributeError at /admin/pages/9/edit/
Exception Value: 'NoneType' object has no attribute 'from_json'`

**Comment by briaguya** [comment]
Did a bit more digging into this (Thanks @Niicck for getting the undocker stuff working) and managed to get to this error:
<img width="811" alt="Screen Shot 2019-06-27 at 1 25 33 AM" src="https://user-images.githubusercontent.com/6384156/60242090-7c477280-987a-11e9-919f-04140ba9b2e5.png">

which is new, because it's not a NoneType anymore. After diving in using a debugger 🎉I found a part of the wagtail code that seems to think our pages are redirects? (https://github.com/wagtail/wagtail/blob/master/wagtail/core/models.py#L574)

This happens both when we try publishing a page imported as draft from the main menu with our custom publish view, as well as when I tried removing our custom publish view (and all references to it so our views don't break) and then using publish from the edit page.
